joshprzybyszewski/cribbage
# React dev upgrades

## What broke / What you're adding
- Use absolute paths instead of relative ones to avoid `../../` hell
- Add linting the client to our github workflows

Node doesn't do _great_ with our root directory not being a node application (with a `package.json`). If you want vscode to use the locally installed eslint, follow these steps
- install the eslint plugin for vscode
- `cd` into `client/` and do an `npm install`
- start vscode from `client/` instead of from our root! (this is a pain but otherwise vscode doesn't pick up anything)

## How you did it
[Absolute paths](https://create-react-app.dev/docs/importing-a-component/#absolute-imports)
[Linting workflow](https://github.com/marketplace/actions/run-eslint)

## How to test it and how to try to break it
_shrug_ run the app, see it still work with new import paths. Push react code up, get rekt by eslint
